src/ORM/DB.php

```diff
@@ -58,31 +75,57 @@ class DB
      */
     protected static $configs = [];
 
-
-
     /**
      * The last SQL query run.
      * @var string
      */
     public static $lastQuery;
 
+    /**
+     * The name of the last connection used. This is only used for unit-testing purposes.
+     * @interal
+     */
+    private static string $lastConnectionName = '';
+
     /**
      * Internal flag to keep track of when db connection was attempted.
      */
     private static $connection_attempted = false;
 
+    /**
+     * Only use the primary database connection for the rest of the current request
+     *
+     * @internal
+     */
+    private static bool $mustUsePrimary = false;
+
+    /**
+     * Used by DB::withPrimary() to count the number of times it has been called
+     * Uses an int instead of a bool to allow for nested calls
+     *
+     * @internal
+     */
+    private static int $withPrimaryCount = 0;
+
+    /**
+     * The key of the replica config to use for this request
+     *
+     * @internal
+     */
+    private static string $replicaConfigKey = '';
+
     /**
      * Set the global database connection.
      * Pass an object that's a subclass of SS_Database.  This object will be used when {@link DB::query()}
      * is called.
      *
      * @param Database $connection The connection object to set as the connection.
      * @param string $name The name to give to this connection.  If you omit this argument, the connection
-     * will be the default one used by the ORM.  However, you can store other named connections to
+     * will be the primary one used by the ORM.  However, you can store other named connections to
      * be accessed through DB::get_conn($name).  This is useful when you have an application that
      * needs to connect to more than one database.
      */
-    public static function set_conn(Database $connection, $name = 'default')
+    public static function set_conn(Database $connection, $name = DB::CONN_PRIMARY)
```

Comment:
```suggestion
    public static function set_conn(Database $connection, $name)
```
Don't have a default arg here, to prevent people trying to use the following an expecting it to work:
```php
DB::set_conn($myConn);
$sameConn = DB::get_conn();
```

Any code like that from cms 5 should explicitly throw "hey you need to pass in a connection name" errors in cms 6, which is easiest to do by removing the default arg.


Keep the default arg in `get_conn()` because that one makes sense.

Reply:
This still hasn't been done or replied to?

Reply:
Updated